model/event_context.go

```diff
@@ -0,0 +1,115 @@
+package model
+
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+
+	"github.com/google/go-github/v45/github"
+	"github.com/mattermost/release-bot/config"
+)
+
+type EventContext interface {
+	GetConclusion() string
+	GetEvent() string
+	GetInstallationID() int64
+	GetName() string
+	GetRepository() string
+	GetStatus() string
+	GetWorkflow() string
+	GetWorkflowRunID() int64
+	GetCommitHash() string
+	GetReference() string
+	IsBranch() bool
+	IsFork() bool
+	IsPr() bool
+	IsTag() bool
+	Log()
+}
+
+type payloadToEventContextConverter func(payload []byte) (EventContext, error)
+
+var eventContextConverters map[string]payloadToEventContextConverter = map[string]payloadToEventContextConverter{
+	"push":         pushEventMapper,
+	"workflow_run": workflowRunEventMapper,
+}
+
+func pushEventMapper(payload []byte) (EventContext, error) {
+	var event github.PushEvent
+	if err := json.Unmarshal(payload, &event); err != nil {
+		return nil, err
+	}
+	return newPushEventContext(event), nil
+}
+
+func workflowRunEventMapper(payload []byte) (EventContext, error) {
+	var event github.WorkflowRunEvent
+	if err := json.Unmarshal(payload, &event); err != nil {
+		return nil, err
+	}
+	return newWorkflowRunEventContext(event), nil
+}
+
+func ConvertPayloadToEventContext(githubEventType string, payload []byte) (EventContext, error) {
+	if converter, ok := eventContextConverters[githubEventType]; ok {
+		return converter(payload)
+	}
+	return nil, fmt.Errorf("converter not found for %s event", githubEventType)
+}
+
+func GetTargetPipeline(context EventContext, pipelines []config.PipelineConfig) *config.PipelineConfig {
+	for _, pipeline := range pipelines {
+		for _, condition := range pipeline.Conditions {
+			var result *config.PipelineConfig = &pipeline
+			if !contains(condition.Webhook, context.GetEvent()) {
```

Comment:
I am a bit confused for the whole block. could you elaborate what exactly you are trying to filter?

Overall using `result==nil` and iterate is not a very good practice.

Reply:
Will try to simplify that logic.

Reply:
That code block is to find a private pipeline matches to the selectors. atm we have about 10 selectors.

Reply:
Code block is simplified and comment added to the code base.

Technically we have a below block at configuration.

```
pipelines:
  - organization: mattermost
    repository: release-pipelines
    workflow: docker.yaml
    conditions:
      - repository: "^mattermost/.*$"
        webhook: [ workflow_run ] 
        workflow: "Build"
        pr: true
        fork: true
        status: "completed"
        conclusion: "success"
      - repository: "^mattermost/.*$"
        webhook: [ workflow_run ]
        workflow: "Build"
        branch: true
        name: "^main$"
        status: "completed"
        conclusion: "success"   
```

Reply:
looks good to me now